generate/swaggergen/g_docs.go

```diff
@@ -695,23 +695,49 @@ func parserComments(f *ast.FuncDecl, controllerName, pkgpath string) error {
 				}
 				opts.Parameters = append(opts.Parameters, para)
 			} else if strings.HasPrefix(t, "@Failure") {
+				ss := strings.TrimSpace(t[len("@Failure"):])
 				rs := swagger.Response{}
-				st := strings.TrimSpace(t[len("@Failure"):])
-				var cd []rune
-				var start bool
-				for i, s := range st {
-					if unicode.IsSpace(s) {
-						if start {
-							rs.Description = strings.TrimSpace(st[i+1:])
-							break
-						} else {
-							continue
+				respCode, pos := peekNextSplitString(ss)
+				ss = strings.TrimSpace(ss[pos:])
+				respType, pos := peekNextSplitString(ss)
+				if respType == "{object}" || respType == "{array}" {
+					isArray := respType == "{array}"
+					ss = strings.TrimSpace(ss[pos:])
+					schemaName, pos := peekNextSplitString(ss)
+					if schemaName == "" {
+						beeLogger.Log.Fatalf("[%s.%s] Schema must follow {object} or {array}", controllerName, funcName)
+					}
+					if strings.HasPrefix(schemaName, "[]") {
+						schemaName = schemaName[2:]
+						isArray = true
+					}
+					schema := swagger.Schema{}
+					if sType, ok := basicTypes[schemaName]; ok {
+						typeFormat := strings.Split(sType, ":")
+						schema.Type = typeFormat[0]
+						schema.Format = typeFormat[1]
+					} else {
+						m, mod, realTypes := getModel(schemaName)
+						schema.Ref = "#/definitions/" + m
+						if _, ok := modelsList[pkgpath+controllerName]; !ok {
+							modelsList[pkgpath+controllerName] = make(map[string]swagger.Schema)
 						}
+						modelsList[pkgpath+controllerName][schemaName] = mod
+						appendModels(pkgpath, controllerName, realTypes)
 					}
-					start = true
-					cd = append(cd, s)
+					if isArray {
+						rs.Schema = &swagger.Schema{
+							Type:  astTypeArray,
+							Items: &schema,
+						}
+					} else {
+						rs.Schema = &schema
+					}
+					rs.Description = strings.TrimSpace(ss[pos:])
+				} else {
+					rs.Description = strings.TrimSpace(ss)
 				}
-				opts.Responses[string(cd)] = rs
+				opts.Responses[respCode] = rs
```

Comment:
can you help to extra this part to be a new function which returns the `swagger.Response`:
```
parseFailureResponse(annotation string) swagger.Response
```
And then you can write unit tests for this function.

I know this repository has no unit tests, but we can improve it

Reply:
Turning this duplication into a standalone function would work, but with the current one-liner and shared memory model I don't think it's possible to write good unit tests.

Reply:
It's not complicate.
You only need to reset the package variable when step in and step out the function, like this:
```go
func xxx() {
   reset()
    defer reset()
}
```

Reply:
@flycash You can take a look at my new commit.

Reply:
I check your commit and it looks good. But I didn't see any tests

Reply:
I have never learn about package "go/ast", so I have no ieda to reset any variable.

Reply:
If you think it is easy, you can finish this unit test.